client/wsclient.go

```diff
@@ -220,12 +309,22 @@ func (c *wsClient) runOneCycle(ctx context.Context) {
 	// Create a cancellable context for background processors.
 	procCtx, procCancel := context.WithCancel(ctx)
 
+	// Stop processors if we receive a signal to do so.
+	go func() {
+		select {
+		case <-c.stopProcessors:
+			procCancel()
+			c.sender.WaitToStop()
+			c.processorsStopped <- struct{}{}
```

Comment:
Maybe add a comment that `stopProcessors` doesn't immediately stop the receiver processor but it is expected to read the last message with a close error and exit.

Reply:
I tried to call this out in other places, but it's also fair to call it out here. I also renamed the struct fields to make their intent more clear. Let me know if this is still unclear.